src/EnergyPlus/Plant/Solution/PlantLoopSolver.cc

```diff
@@ -168,215 +158,248 @@ namespace PlantLoopSolver {
-		using PlantUtilities::TightenNodeMinMaxAvails;
-
-		// Locals
-		// SUBROUTINE ARGUMENT DEFINITIONS:
-
-		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-		//~ Topology variables
-		int ThisSideInletNode; // Plant loop side loop inlet
-		int ThisSide;
-		int OtherSide;
-
-		//~ Initialization and validation flags
-
-		//~ Flags
-		bool LoopShutDownFlag;
-
-		//~ Other variables
-		Real64 ThisLoopSideFlow;
-		Real64 TotalPumpMaxAvailFlow;
-		Real64 TotalPumpMinAvailFlow;
-
-		// Object Data
-		m_FlowControlValidator IsLoopSideValid;
-
 		// Initialize variables
 		InitialDemandToLoopSetPoint = 0.0;
 		CurrentAlterationsToDemand = 0.0;
 		UpdatedDemandToLoopSetPoint = 0.0;
-		ThisSide = LoopSideNum;
-		OtherSide = 3 - ThisSide; //will give us 1 if thisside is 2, or 2 if thisside is 1
-		LoopShutDownFlag = false;
-		ThisSideInletNode = PlantLoop( LoopNum ).LoopSide( ThisSide ).NodeNumIn;
+		int ThisSide = LoopSideNum;
+		int OtherSide = 3 - ThisSide; //will give us 1 if thisside is 2, or 2 if thisside is 1
+
+		auto & thisPlantLoop = DataPlant::PlantLoop( LoopNum );
+		auto & thisLoopSide = thisPlantLoop.LoopSide( ThisSide );
+		auto & otherLoopSide = thisPlantLoop.LoopSide( OtherSide );
+
+		int ThisSideInletNode = thisLoopSide.NodeNumIn;
 
 		// The following block is related to validating the flow control paths of the loop side
 		// Since the control types are scheduled, I think BeginTimeStep should be a decent check frequency
-		if ( BeginTimeStepFlag && PlantLoop( LoopNum ).LoopSide( ThisSide ).OncePerTimeStepOperations ) {
+		if ( DataGlobals::BeginTimeStepFlag && thisLoopSide.OncePerTimeStepOperations ) {
 
 			// Initialize loop side controls -- could just be done for one loop since this routine inherently
 			//  loops over all plant/condenser loops.  Not sure if the penalty is worth investigating.
-			InitLoadDistribution( FirstHVACIteration );
+			PlantCondLoopOperation::InitLoadDistribution( FirstHVACIteration );
 
 			// Now that the op scheme types are updated, do LoopSide validation
-			IsLoopSideValid = ValidateFlowControlPaths( LoopNum, ThisSide );
+			auto IsLoopSideValid = thisPlantLoop.loopSolver.ValidateFlowControlPaths( LoopNum, ThisSide );
 			if ( ! IsLoopSideValid.Valid ) {
 				ShowFatalError( "ERROR:" + IsLoopSideValid.Reason );
 			}
 
 			// Set the flag to false so we won't do these again this time step
-			PlantLoop( LoopNum ).LoopSide( ThisSide ).OncePerTimeStepOperations = false;
+            thisLoopSide.OncePerTimeStepOperations = false;
 
 		} else {
 
 			// Set the flag to true so that it is activated for the next time step
-			PlantLoop( LoopNum ).LoopSide( ThisSide ).OncePerTimeStepOperations = true;
+            thisLoopSide.OncePerTimeStepOperations = true;
 
 		}
 
 		// Do pressure system initialize if this is the demand side (therefore once per whole loop)
-		if ( ThisSide == DemandSide ) SimPressureDropSystem( LoopNum, FirstHVACIteration, PressureCall_Init );
```

Comment:
These changes are hard to dig through, but it's just organizational.  Logic was pulled into tighter functions to make it easy to repeat the call to the solution.  I'll highlight the actual functional change below.